internal/app/maintenance/server.go

```diff
@@ -63,15 +64,25 @@ func (s *Server) Register(obj *grpc.Server) {
 
 	// wrap resources with access filter
 	resourceState := s.controller.Runtime().State().V1Alpha2().Resources()
-	resourceState = state.WrapCore(state.Filter(resourceState, resources.AccessPolicy(resourceState)))
+
+	// If the peer is a siderolink peer, allow access to all resources.
+	accessPolicyFilter := func(ctx context.Context, access state.Access) error {
```

Comment:
do we even need that given that the the siderolink user will have admin role injected?

Reply:
nope, I reverted this part, as I changed the approach.